### PR TITLE
[Android] Fix shields button is not clickable when shields is off

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/6032

Shields button should be clickable always.
This bug is regression from controlling enable/disable shields button.
To fix this, shields button should be always enabled.

The purpose of disabling shields button was preventing shields button
click when tab is not ready.
So, instead of disabling shields button, just return when current tab is
null during the handling of click.

And enable/disableShieldsButton methods are integrated into
updateBraveShieldsButtonState().

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java [hunk 1]
```diff
@@ -186,34 +165,18 @@ public void onClick(View v) {
           }
           Tab currentTab = mMainActivity.getActivityTab();
           if (currentTab == null) {
-              assert false;
               return;
           }
           try {
               URL url = new URL(currentTab.getUrl());
-              updateBraveShieldsButtonState(currentTab.getProfile(), url.toString());
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java]: We don't need to update shields button state while click event handling.

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java [hunk 2]
```diff
@@ -186,34 +165,18 @@ public void onClick(View v) {
...
               mBraveShieldsMenuHandler.show(mBraveShieldsButton, url.toString(),
                   url.getHost(), currentTab.getId(), currentTab.getProfile());
           } catch (Exception e) {
-              disableBraveShieldsButton();
+              // Do nothing. Just return w/o showing shields popup.
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java]: in that case we will show enabled button on URLs like `chrome://flags` and etc

Comment (simonhong) [reply android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java]: I checked that `isShieldsOnForTab()` returns false properly for those urls.
Also I added more check for protocols by (`isInvalidProtocolForShields`).
In desktop, shields are disabled for some protocols like `ftp` or `file` protocols.
I think android brave also should do like desktop.

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayout.java]: ok

**Pull request merged**